pkg/util/log/clog.go

```diff
@@ -930,6 +933,18 @@ func (sb *syncBuffer) rotateFile(now time.Time) error {
 		return err
 	}
 
+	// Redirect stderr to the current INFO log file in order to capture panic
+	// stack traces that are written by the Go runtime to stderr. Note that if
+	// --logtostderr is true we'll never enter this code path and panic stack
+	// traces will go to the original stderr as you would expect.
+	if sb.sev == Severity_INFO &&
+		logging.stderrThreshold > Severity_INFO &&
+		!logging.noStderrRedirect {
+		if err := syscall.Dup2(int(sb.file.Fd()), syscall.Stderr); err != nil {
```

Comment:
Regarding the "graceful shutdown failed" Arguably I think this particular message could also be duplicated on the original stderr, in addition to the log file.